### ISSUE TITLE
provided hook

### ISSUE THREAD

**Issue by GaborTorma** [nuxt/nuxt#23497]
### Describe the feature

Please create a new hook, that called when new value provided.
Extend your [provide](https://github.com/nuxt/nuxt/blob/6ce94a07623d2520f45668a23c886ed0b171d056/packages/nuxt/src/app/nuxt.ts#L268) function with this row:
`nuxtApp.callHook('provided:' + name, value)`

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by huang-julien** [comment]
Hi :wave: do you have an exemple of how you wish to use this ?

Hookable is extensible meaning you can also implement this on your project

**Comment by GaborTorma** [comment]
Yes, I know the hooks is extensible, but if I want to extend the another module with my module, I need to know when the original module provided the value. If nuxt will have `provided` hook we will have an universal way to handle this situations.

For examples:

- I extend [dayjs-nuxt](https://nuxt.com/modules/dayjs) modul with i18n support via my [nuxt-dayjs-i18n](https://github.com/GaborTorma/nuxt-dayjs-i18n) modul. I only extend dayjs when the dayjs-nuxt modul prodvided dayjs instance. I resolved it with [`app:created`](https://github.com/GaborTorma/nuxt-dayjs-i18n/blob/main/src/runtime/plugin.ts#L7) hook, but `provided:dayjs` hook would be better.

- I would extend [vuetify-nuxt-module](https://nuxt.com/modules/vuetify-nuxt-module) with [vuetify3-dialog](https://github.com/ThomasLeconte/vuetify3-dialog) plugin via my nuxt-vuetify3-dialog module. I [need the vueitfy instance for configuring](https://github.com/ThomasLeconte/vuetify3-dialog#:~:text=vuetify%3A%20vuetifyInstance) vuetify3-dialog plugin, but the vuetify instance does not provieded before `app:created` hook. I made a [PR for vuetify-nuxt-module for vuetify:created hook](https://github.com/userquin/vuetify-nuxt-module/pull/112), but to use `provided:vuetify` hook would be the universal and easier way.
